File: 007/007.py
class Solution:
    def reverse(self, x):
        
        ## Input:
        # x: int
        ## Output:
        # int

        if (x < 0):
            isNegative = True
        else:
            isNegative = False
        
        x = abs(x)
        rev_x = 0
        while (x >= 1):
           rev_x = 10 * rev_x + x % 10
           if ((isNegative and rev_x > 2 ** 31) or (not isNegative and rev_x >= 2 ** 31)):
               return 0
           x = x // 10
        
        if (isNegative):
            return -rev_x
        else:
            return rev_x      

File: 007/007.cpp
class Solution
{
public:
    int reverse(int x)
    {
        bool isNegative = false;
        if (x < 0)
        {
            isNegative = true;
        }

        long y = abs(x);
        long rev_y = 0;
        while (y >= 1)
        {
            rev_y = 10 * rev_y + y % 10;
            if ((isNegative && rev_y > pow(2, 31)) || (!isNegative && rev_y >= pow(2, 31)))
            {
                return 0;
            }
            y /= 10;
        }

        if (isNegative)
        {
            return int(-rev_y);
        }
        else
        {
            return int(rev_y);
        }
    }
};